src/main/java/com/cloudbees/hudson/plugins/folder/ChildNameGenerator.java

```diff
@@ -185,6 +194,73 @@ protected final String idealNameFromItem(@NonNull P parent, @NonNull I item) {
     @CheckForNull
     public abstract String dirNameFromItem(@NonNull P parent, @NonNull I item);
 
+    public static class ResultWithOptionalSave<I> {
```

Comment:
Can this and `ensureItemDirectory` be restricted to internal use, or package-scope? (`ChildNameGenerator` as a whole is used by `branch-api`.)